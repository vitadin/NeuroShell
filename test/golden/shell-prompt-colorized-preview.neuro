%% Colorized shell prompt preview test
\set[_echo_command="true"]
%% Tests that shell-prompt-preview properly renders colors instead of showing raw markup

%% Set up context variables for testing
\set[custom_var="test_value"]

%% Test preview with minimal-color preset
\shell-prompt-preset[style="minimal-color"]
\shell-prompt-preview

%% Test preview with custom color markup
\set[_prompt_line1={{color:red}}ERROR{{/color}} {{bold}}BOLD{{/bold}} {{color:green}}OK{{/color}}> ]
\shell-prompt-preview

%% Test preview with semantic colors and variables
\set[_prompt_line1={{color:info}}${@user}{{/color}}@{{color:success}}${@pwd}{{/color}} [{{color:warning}}${custom_var}{{/color}}] ]
\set[_prompt_line2={{color:cyan}}❯{{/color}} ]
\set[_prompt_lines_count=2]
\shell-prompt-preview

%% Test preview with style markup combinations
\set[_prompt_line1={{bold}}{{color:blue}}BOLD-BLUE{{/color}}{{/bold}} {{italic}}{{color:yellow}}ITALIC-YELLOW{{/color}}{{/italic}}> ]
\set[_prompt_lines_count=1]
\shell-prompt-preview