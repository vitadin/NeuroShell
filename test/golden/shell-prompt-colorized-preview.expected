Setting _echo_command = true
%%> "\\set[custom_var=\"test_value\"]"
Setting custom_var = test_value
%%> "\\shell-prompt-preset[style=\"minimal-color\"]"
Applied minimal-color prompt preset
%%> "\\shell-prompt-preview"
Prompt Preview:
===============
neuro> 
█
%%> "\\set[_prompt_line1={{color:red}}ERROR{{/color}} {{bold}}BOLD{{/bold}} {{color:green}}OK{{/color}}> ]"
Setting _prompt_line1 = {{color:red}}ERROR{{/color}} {{bold}}BOLD{{/bold}} {{color:green}}OK{{/color}}>
%%> "\\shell-prompt-preview"
Prompt Preview:
===============
ERROR BOLD OK>
█
%%> "\\set[_prompt_line1={{color:info}}${@user}{{/color}}@{{color:success}}${@pwd}{{/color}} [{{color:warning}}${custom_var}{{/color}}] ]"
Setting _prompt_line1 = {{color:info}}testuser{{/color}}@{{color:success}}/test/pwd{{/color}} [{{color:warning}}test_value{{/color}}]
%%> "\\set[_prompt_line2={{color:cyan}}❯{{/color}} ]"
Setting _prompt_line2 = {{color:cyan}}❯{{/color}}
%%> "\\set[_prompt_lines_count=2]"
Setting _prompt_lines_count = 2
%%> "\\shell-prompt-preview"
Prompt Preview:
===============
testuser@/test/pwd [test_value]
❯
█
%%> "\\set[_prompt_line1={{bold}}{{color:blue}}BOLD-BLUE{{/color}}{{/bold}} {{italic}}{{color:yellow}}ITALIC-YELLOW{{/color}}{{/italic}}> ]"
Setting _prompt_line1 = {{bold}}{{color:blue}}BOLD-BLUE{{/color}}{{/bold}} {{italic}}{{color:yellow}}ITALIC-YELLOW{{/color}}{{/italic}}>
%%> "\\set[_prompt_lines_count=1]"
Setting _prompt_lines_count = 1
%%> "\\shell-prompt-preview"
Prompt Preview:
===============
BOLD-BLUE ITALIC-YELLOW>
█